src/Analyzer.Cli/CommandLineParser.cs

```diff
@@ -198,10 +198,56 @@ private void SetupCommonOptionsForCommands(List<Command> commands)
                 return (int)ExitCode.ErrorInvalidARMTemplate;
             }
 
-            var analysisResult = AnalyzeTemplate(templateFilePath, parametersFilePath);
+            ExitCode exitCode;
+            if (parametersFilePath != null)
+            {
+                exitCode = AnalyzeTemplate(templateFilePath, parametersFilePath);
+            }
+            else
+            {
+                // Check if a parameters.json file is present according to naming standards here https://learn.microsoft.com/en-us/azure/azure-resource-manager/templates/parameter-files#file-name, and if so use it as the parametersFile input
+                var parametersFiles = FindParameterFileForTemplate(templateFilePath);
+                int numOfFilesAnalyzed = 0;
+                bool issueReported = false;
+                var filesFailed = new List<FileInfo>();
+
+                if (parametersFiles.Count() > 0)
+                {
+                    foreach (FileInfo parametersFile in parametersFiles)
```

Comment:
I see that most of the code inside this `if` was copied from the AnalyzeDirectory function, could you abstract both blocks into a function to avoid code duplication?